src/behavior-considered-undefined.md

```diff
@@ -42,9 +42,14 @@ code.
   All this also applies when values of these
   types are passed in a (nested) field of a compound type, but not behind
   pointer indirections.
-* Mutating immutable data. All data inside a [`const`] item is immutable. Moreover, all
-  data reached through a shared reference or data owned by an immutable binding
-  is immutable, unless that data is contained within an [`UnsafeCell<U>`].
+* Mutating immutable bytes. All bytes inside a [`const`] item are immutable.
+  The bytes owned by an immutable binding are immutable, unless those bytes are part of an [`UnsafeCell<U>`].
+
+  Moreover, the bytes [pointed to] by a shared reference, including transitively through other references (both shared and mutable) and `Box`es, are immutable: transitivity includes those references stored in fields of compound types.
```

Comment:
The nuance of `;` vs `:` is rather annoying - I used `:` here because the second main clause is a clarification that would struggle to stand as a sentence (I'd start it with "This transitivity" if it was a fresh sentence.) Both can be used here, so it's really just a question of style.